tests/thread/thread_stdin.py

```diff
@@ -0,0 +1,42 @@
+# test that having multiple threads block on stdin doesn't cause any issues.
+#
+# The test doesn't expect any input on stdin.
+#
+# This is a regression test for https://github.com/micropython/micropython/issues/15230
+# on rp2, but doubles as a general property to test across all ports.
+import select
+import sys
+import time
+import _thread
+
+
+class StdinWaiter:
+    def __init__(self):
+        self._done = False
+
+    def wait_stdin(self, timeout_ms):
+        poller = select.poll()
+        poller.register(sys.stdin, select.POLLIN)
+        poller.poll(timeout_ms)
+        # ignoring the poll result as we don't expect any input
+        self._done = True
+
+    def is_done(self):
+        return self._done
+
+
+thread_waiter = StdinWaiter()
+_thread.start_new_thread(thread_waiter.wait_stdin, (1000,))
+StdinWaiter().wait_stdin(1000)
+
+# add a short delay to account for inconsistency waking the two threads (on
+# CPython CI runners there seems to be some intermittent difference in thread
+# wakeup delays.) On Linux we expect both threads should resume more or less
+# concurrently - aka the "thundering herd" effect"
+try:
+    time.sleep_ms(100)
+except AttributeError:
+    time.sleep(0.1)
```

Comment:
How about just doing `time.sleep(0.1)` without any try-except?  All ports should support that.

Reply:
Oh, I somehow had it in my head for many years that there was a MicroPython configuration that didn't support floats.

Easy!

Reply:
You are right, there are configs without float.  CC3200 is the main example that has threading but no float, and these thread tests were originally written to run on that target.

It's probably worth retaining that ability, ie don't use floats here.

But note that if floats are enabled on a build, then even parsing `0.1` will raise a `SyntaxError` before the code runs.  So your original code won't support non-float builds.

How about just a busy loop waiting for the done flag to be set?  (Also don't want a dependency on `machine` here.)  The test running will time out after 30 seconds if the flag is never set.

Reply:
Have changed as suggested.